src/pspm_data_editor.m

```diff
@@ -838,7 +841,13 @@ function UpdateEpochList(hObject)
     set(handles.lbEpochs, 'Value', sel_ep);
   end
   set(handles.lbEpochs, 'String', names);
-  handles.epochs = epochs;
+  % update to sort epochs according to the time series order
+  epochs_list = 1:length(epochs);
+  for i_epochs_list = 1:length(epochs) % obtain data
+    epochs_list(i_epochs_list) = epochs{1,i_epochs_list}.range(1);
+  end
+  [~,idx]=sort(epochs_list);
+  handles.epochs = epochs{1,idx};
```

Comment:
FYI I think the six previous lines can be shorten as:
`handles.epochs = sort( cellfun(@(epoch) epoch.range(1), epochs) )`
Which might be less intuitive to read or not. I let you be the judge of that

Reply:
Great idea, thanks Jules!

Reply:
Oh? There might be a typo from my end then.
What is the output of `cellfun(@(epoch) epoch.range(1), epochs)`?
Alternatively, can you send me be email (jules.brochard@gmail.com) an example of the variable `epochs`?

Reply:
> Oh? There might be a typo from my end then.
> What is the output of `cellfun(@(epoch) epoch.range(1), epochs)`?
> Alternatively, can you send me be email ([jules.brochard@gmail.com](mailto:jules.brochard@gmail.com)) an example of the variable `epochs`?

No problem, here are two screenshots showing the results... I will send you the link to the test files soon.
original solution
<img width="774" alt="Screenshot 2022-06-03 at 16 48 17" src="https://user-images.githubusercontent.com/3895146/171900654-b4ebd5f0-0f06-4d81-badd-962fdb06b0d8.png">
proposed solution
<img width="773" alt="Screenshot 2022-06-03 at 16 47 28" src="https://user-images.githubusercontent.com/3895146/171900700-9c3f2028-114e-4a1b-9507-4980caf030f9.png">

Reply:
it is strange indeed.

What is the output of `cellfun(@(epoch) epoch.range(1), epochs)`? 
This should simply extract the .range(1) field of every element of the cell array `epochs`.

Could you also try `cellfun(@(epoch) epoch.range(1), epochs(1,:))` ?